apps/client/app/components/PurchaseOrderForm.tsx

```diff
@@ -0,0 +1,172 @@
+  vendor_name: Yup.string().required('Vendor name is required'),
+  order_date: Yup.string().required('Order date is required'),
+  expected_delivery_date: Yup.string().required('Expected delivery date is required').test(
+    'is-greater',
+    'Expected delivery date must be after the order date',
+    function (value) {
+      const { order_date } = this.parent;
+      return new Date(order_date) < new Date(value);
+    }
+  ),
+  purchase_order_line_items: Yup.array().of(
+    Yup.object().shape({
+      id: Yup.number().required('Purchase Order Line Item is required'),
+      item_id: Yup.number().required('Item is required').positive(),
+      quantity: Yup.number().required('Quantity is required').positive().integer(),
+      unit_cost: Yup.number().required('Unit cost is required').positive(),
+    })
+  ).required(),
+  total_quantity: Yup.number().notRequired(),
+  total_cost: Yup.number().notRequired(),
+});
+
+const PurchaseOrderForm: React.FC<PurchaseOrderFormProps> = ({ purchaseOrder, items, mode }) => {
+  const [serverErrors, setServerErrors] = useState<string[]>([]);
+
+  // Initialize the form with react-hook-form
+  const { control, handleSubmit, formState: { errors } } = useForm<PurchaseOrder>({
+    defaultValues: {
+      id: purchaseOrder?.id || 0,
+      vendor_name: purchaseOrder?.vendor_name || '',
+      order_date: purchaseOrder?.order_date.substring(0, 10) || '',
+      expected_delivery_date: purchaseOrder?.expected_delivery_date.substring(0, 10) || '',
+      purchase_order_line_items: purchaseOrder?.purchase_order_line_items.map(lineItem => ({
+        id: lineItem.id || 0,
+        item_id: lineItem.item_id,
+        quantity: lineItem.quantity,
+        unit_cost: lineItem.unit_cost,
+      })) || [{ id: 0, item_id: 0, quantity: 0, unit_cost: 0 }]
+    },
+    resolver: yupResolver(validationSchema) as any, // Casting to any to get rid of TypeScript error that is not actually an issue
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'purchase_order_line_items'
+  });
+
+  // On form submission, either call the api that will create a Purchase Order or the api that will update a Purchase Order
+  const onSubmit = async (data: Omit<PurchaseOrder, 'id'>) => {
+    const url = mode === 'create'
+      ? 'http://localhost:3100/api/purchase-orders'
+      : `http://localhost:3100/api/purchase-orders/${purchaseOrder?.id}`;
+    const method = mode === 'create' ? 'POST' : 'PATCH';
+
+    try {
+      const response = await fetch(url, {
+        method,
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(data),
+      });
+
+      if (!response.ok) {
+        const errorData = await response.json();
+        setServerErrors(errorData.message || ['An error occurred']);
+      } else {
+        // Navigate to the list page and reload to ensure server-side rendering takes place
+        window.location.href = '/purchase-orders';
```

Comment:
better to use nextjs's router here. setting location href will re-download js files if not cached and lose client state

Reply:
I wanted to kick off a server reload so that the list page reflects the record that was just added. My understanding is that using the nextjs router would not accomplish that. Maybe I'm missing something simple?

Reply:
I was mistaken. Using the nextjs router works. Thanks!